api/swagger.yaml

```
@@ -4611,9 +4611,9 @@ paths:
       parameters:
         - name: "name"
           in: "query"
-          description: "Assign the specified name to the container. Must match `/?[a-zA-Z0-9_-]+`."
+          description: "Assign the specified name to the container. Must match `[a-zA-Z0-9][a-zA-Z0-9_.-]+`."
           type: "string"
-          pattern: "/?[a-zA-Z0-9_-]+"
+          pattern: "[a-zA-Z0-9][a-zA-Z0-9_.-]+"
```

Wondering if this needs to be capped to the start/end (`^...$`), otherwise it would do partial matches; https://swagger.io/docs/specification/data-models/data-types/#pattern

```suggestion
          pattern: "^[a-zA-Z0-9][a-zA-Z0-9_.-]+$"
```

Looking at the original RegEx, that one seems to take the optional leading `/` into account (`/?`). Combined with the above;

```suggestion
          pattern: "^/?[a-zA-Z0-9][a-zA-Z0-9_.-]+$"
```

---

I _think_ the leading `/` is stripped before we hit the RegEx in the code, but you might want to double-check if we actually accept the leading `/` in the API (that slash was there for historic reasons, so not sure if it's still there)

---

Yup, can confirm that we still have it.
https://github.com/moby/moby/blob/8e610b2b55bfd1bfa9436ab110d311f5e8a74dcb/daemon/names.go#L59-L64